loglib/qlog.c

```diff
@@ -382,12 +379,19 @@ int qlog_packet_frame(bytestream * s, void * ptr)
         if ((ftype & 4) != 0) {
             byteread_vint(s, &offset);
         }
-        uint64_t length = bytestream_remain(s);
-        if ((ftype & 2) != 0) {
-            byteread_vint(s, &length);
-        }
+        uint64_t length = 0;
+        byteread_vint(s, &length);
         fprintf(f, ", \"id\": %"PRIu64", \"offset\": %"PRIu64", \"length\": %"PRIu64", \"fin\": %s ",
             stream_id, offset, length, (ftype & 1) ? "true":"false");
+        if ((ftype & 2) == 0) {
```

Comment:
Wouldn't it be better to only write the length (and offset) fields into the qlog when they were set in the frame header? I.e. no need for "has_length" / "has_offset".

Reply:
See above.